runelite-client/src/main/java/net/runelite/client/plugins/wiki/WikiSearchChatboxTextInput.java

```diff
@@ -297,7 +299,14 @@ public void keyPressed(KeyEvent ev)
 
 	private void search(String search)
 	{
-		LinkBrowser.browse(WikiPlugin.WIKI_BASE + "?search=" + URLEncoder.encode(search) + "&" + WikiPlugin.UTM_PARAMS);
+		try
+		{
+			LinkBrowser.browse(WikiPlugin.WIKI_BASE + "?search=" + URLEncoder.encode(search, Charset.defaultCharset().name()) + "&" + WikiPlugin.UTM_PARAMS);
```

Comment:
you have just inlined the deprecated method. Try reading the deprecation notice as to why this is wrong.

Reply:
I have not inlined the deprecated method. The deprecated method used here was `encode(String s)`. The change I committed uses the method in the deprecation notice, `encode(String s, String enc`.

Reply:
the way you have used `encode(String, String)` is exactly what `encode(String)` does. You have inlined it. It is incorrect for the same reason that `encode(String)` is deprecated.

Reply:
Use HttpUrl.build or HttpUrl.parse

Reply:
Yes, they do the same thing, but `encode(String, String)` is not deprecated, whereas `encode(String)` is. That's the point. However, if you would prefer I change the code to use the HttpUrl class, I will do so.

Reply:
Things are not deprecated arbitrarily. calling `encode(String, Charset.defaultCharset().name())` is the same as `encode(String)`, but without a warning. The bug that the warning warns you about still exists. IMO your way is worse because it hides a valid bug. That being said, the correct thing to do is use HttpUrl.Builder, so do that

Reply:
Commit e46a69d now uses `HttpUrl.parse`.